assetripper: init at 1.1.3

diadatp opened the pull request:
## Description of changes

AssetRipper is a tool for extracting assets from Unity serialized files (CAB-\*, \*.assets, \*.sharedAssets, etc.) and asset bundles (\*.unity3d, \*.bundle, etc.) and converting them into the native Unity engine format.

https://github.com/AssetRipper/AssetRipper

The patch is is a temporary measure until https://github.com/AssetRipper/AssetRipper/issues/1414 is resolved.

Closes #297634

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
assetripper: init at 1.1.3

UlyssesZh commented:
Works for me. Thank you.

UlyssesZh commented on pkgs/by-name/as/asset-ripper/package.nix:
```diff
@@ -0,0 +1,65 @@
+  src = fetchFromGitHub {
+    owner = "AssetRipper";
+    repo = "AssetRipper";
+    rev = "refs/tags/${finalAttrs.version}";
```
Please rebase on master branch and do this:
```suggestion
    tag = finalAttrs.version;
```
See #355973.

YoshiRulz replied:
I've tried rebasing locally and it doesn't build. I think something the fault is in Nixpkgs, but I haven't finished bisecting.

diadatp replied:
Could you share the build error you are receiving? If it is related to missing lib deps in two of the nuget precompiled libraries, then I have overcome that by including `stdenv.cc.cc.lib` in the `buildInputs` as a temporary fix. Would be interested in finding the breaking commit.

YoshiRulz replied:
That was it, yeah.
```
searching for dependencies of /nix/store/y5y38bj32dd71ki3q1k0slnm2jals223-AssetRipper-1.1.3/lib/AssetRipper/libTexture2DDecoderNative.so
    libstdc++.so.6 -> not found!
    libgcc_s.so.1 -> not found!
auto-patchelf: 2 dependencies could not be satisfied
error: auto-patchelf could not satisfy dependency libstdc++.so.6 wanted by /nix/store/y5y38bj32dd71ki3q1k0slnm2jals223-AssetRipper-1.1.3/lib/AssetRipper/libTexture2DDecoderNative.so
error: auto-patchelf could not satisfy dependency libgcc_s.so.1 wanted by /nix/store/y5y38bj32dd71ki3q1k0slnm2jals223-AssetRipper-1.1.3/lib/AssetRipper/libTexture2DDecoderNative.so
auto-patchelf failed to find all the required dependencies.
Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
```
Culprit being 687371ab28358ca878a93f4ef4884bfecf0ac40c (not sure how removing Python from propagated deps means removing `libstdc++` but there you go).

UlyssesZh commented on pkgs/by-name/as/asset-ripper/package.nix:
```diff
@@ -0,0 +1,65 @@
+
+  nugetDeps = ./deps.nix;
+
+  executables = [ "AssetRipper.GUI.Free" ];
```
This executable name is a bit too unconventional to be used as the name of the main program. Maybe use `ln -s` to create an executable at `$out/bin/AssetRipper` that links to the built executable instead?

YoshiRulz replied:
`assetripper-gui`? There's no CLI in upstream, but I have [a small patch](https://github.com/YoshiRulz/AssetRipper/commit/c6221ea0d73585eb538f850f805d68ee1558b5de.patch) to add one, was going to suggest that for Nixpkgs.

UlyssesZh replied:
I think you should open a PR to upstream instead. Packaging should not add new features, especially such a significant one.

YoshiRulz replied:
AssetRipper/AssetRipper#1565

diadatp replied:
Is the new executable name intuitive? I can resolve this conversation if it is.

UlyssesZh replied:
I was thinking setting `executables = [];` and linking to `$out/lib/${pname}/AssetRipper.GUI.Free` instead. Also technically this step should go in install phase instead of fixup phase. Also `meta.mainProgram` needs changing.

diadatp replied:
My rationale for leaving `executables` as is to allow for the fixup phase hook to wrap the binary here:
https://github.com/NixOS/nixpkgs/blob/a01ee2f7f3e0f33f6738b6f49d24d3653a415682/pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-fixup-hook.sh#L84

If I set `executables = [];`, the main program does get generated in `lib` but is unable find the dotnet runtime when run from there. I am pushing a new commit that shifts the symlink-ing to `postInstall` and changes mainProgram.
Is setting `postInstall` the correct practice?

UlyssesZh replied:
Ah sorry didn't know that you cannot just launch from `lib`. `postInstall` seems correct.

UlyssesZh commented on pkgs/by-name/as/asset-ripper/package.nix:
```diff
@@ -0,0 +1,65 @@
+
+buildDotnetModule (finalAttrs: {
+  pname = "AssetRipper";
+  version = "1.1.3";
```
AssetRipper has new releases:
```suggestion
  version = "1.1.7";
```

YoshiRulz replied:
Need to bump to .NET 9 for this by the way.

YoshiRulz replied:
I tried updating, but `fetch-deps` wasn't working, and in fact it's not working for 1.1.3 i.e. at cdfc9705b45c8d707d1b39b52df04dee1779ca0b without changes. How did you generate `deps.nix`?

diadatp replied:
Good question. The passthru fetch-deps fails for me too. Looking through my shell history, I dropped into a dotnet sdk nix-shell within the project's source and ran approximately the following:

```shell
dotnet restore --packages out
nuget-to-nix out > deps.nix
```

UlyssesZh replied:
Note that in comments. Failing to fetch deps automatically would mean that it cannot be updated properly with an update script, and people have to manually handle that, and they need to know how to do that manually by reading the relevant comments.

I am suspecting this be some bug of nuget-to-nix, but considering that currently there is work going on to retire nuget-to-nix (#362278), it may not worth digging into it. However, it would still be good if you can look into this.

YoshiRulz replied:
I got 1.1.7 building. https://github.com/YoshiRulz/nixpkgs/commit/8ab183d027e177e53f233f673dc4797a443fe28f

UlyssesZh commented on pkgs/by-name/as/asset-ripper/package.nix:
```diff
@@ -0,0 +1,65 @@
+    license = lib.licenses.gpl3Only;
+    mainProgram = "AssetRipper.GUI.Free";
+    maintainers = with lib.maintainers; [ diadatp ];
+    platforms = lib.platforms.unix;
```
```suggestion
    platforms = lib.platforms.unix;
    sourceProvenance = with lib.sourceTypes; [
      fromSource
      binaryNativeCode # library fetched by NuGet
    ];
```

UlyssesZh commented:
Make sure your commit complies with the [relevant conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions). To elaborate, the package name should be all lower-case, and remove the space between "Asset Ripper" in your commit message.

diadatp commented:
Above push does the following:

Updated meta.mainProgram to reflect symlink.
Fixed bug introduced in fixup due to pname change.
Moved symlink creation to postInstall.

UlyssesZh commented:
The "needs upstream fix" tag is not related to nuget-to-nix but is referring to AssetRipper/AssetRipper#1414 (which necessitates the patch in this PR), but yeah, the tag should be removed because that issue is also resolved.

This PR still needs some work: it need to remove the now-unnecessary patch file, and update to the current latest upstream version (and thus .NET 9).

YoshiRulz commented:
[Rebased and updated](https://github.com/YoshiRulz/nixpkgs/commit/c47a126cd3640eba172e74e4a1c3a908b3cff8a7)

UlyssesZh commented:
@YoshiRulz Because @diadatp does not seem to be responding, could you please open a new PR?
